src/zenml/integrations/whylogs/data_validators/whylogs_data_validator.py

```diff
@@ -0,0 +1,176 @@
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the whylogs data validator."""
+
+import datetime
+from typing import Any, ClassVar, Optional, Sequence, cast
+
+import pandas as pd
+import whylogs as why  # type: ignore
+from whylogs.api.writer.whylabs import WhyLabsWriter  # type: ignore
+from whylogs.core import DatasetProfileView  # type: ignore
+
+from zenml.data_validators import BaseDataValidator
+from zenml.environment import Environment
+from zenml.integrations.whylogs import WHYLOGS_DATA_VALIDATOR_FLAVOR
+from zenml.integrations.whylogs.secret_schemas.whylabs_secret_schema import (
+    WhylabsSecretSchema,
+)
+from zenml.logger import get_logger
+from zenml.repository import Repository
+from zenml.steps import STEP_ENVIRONMENT_NAME, StepEnvironment
+
+logger = get_logger(__name__)
+
+
+class WhylogsDataValidator(BaseDataValidator):
+    """Whylogs data validator stack component.
+
+    Attributes:
+        whylabs_secret: Optional ZenML secret with Whylabs credentials. If
+            configured, all the data profiles returned by all pipeline steps
+            will automatically be uploaded to Whylabs in addition to being
+            stored in the ZenML Artifact Store.
+    """
+
+    whylabs_secret: Optional[str] = None
+
+    # Class Configuration
+    FLAVOR: ClassVar[str] = WHYLOGS_DATA_VALIDATOR_FLAVOR
+    NAME: ClassVar[str] = "whylogs"
+
+    def data_profiling(
+        self,
+        dataset: pd.DataFrame,
+        comparison_dataset: Optional[pd.DataFrame] = None,
+        profile_list: Optional[Sequence[str]] = None,
+        dataset_timestamp: Optional[datetime.datetime] = None,
+        **kwargs: Any,
+    ) -> DatasetProfileView:
+        """Analyze a dataset and generate a data profile with whylogs.
+
+        Args:
+            dataset: Target dataset to be profiled.
+            comparison_dataset: Optional dataset to be used for data profiles
+                that require a baseline for comparison (e.g data drift profiles).
+            profile_list: Optional list identifying the categories of whylogs
+                data profiles to be generated (unused).
+            dataset_timestamp: timestamp to associate with the generated
+                dataset profile (Optional). The current time is used if not
+                supplied.
+            **kwargs: Extra keyword arguments (unused).
+
+        Returns:
+            A whylogs profile view object.
+        """
+        results = why.log(pandas=dataset)
+        profile = results.profile()
+        dataset_timestamp = dataset_timestamp or datetime.datetime.utcnow()
+        profile.set_dataset_timestamp(dataset_timestamp=dataset_timestamp)
+        return profile.view()
+
+    def _get_whylabs_secret(self) -> Optional[WhylabsSecretSchema]:
```

Comment:
Just one  comment, not sure if it's applicable here:
This replicates (almost identically) the code that we have in the `zenml.stack.AuthenticationMixin` class.

By inheriting from that class you could get the same functionality without writing any code:
```python
class WhylogsDataValidator(BaseDataValidator, AuthenticationMixin):
    def use_secret(self):
        secret = self.get_authentication_secret(expected_schema_type=WhylabsSecretSchema)
```

The downside:
- The secret property is called `authentication_secret` instead of `whylogs_secret`
- The error messages are a little less customized

Reply:
Nice ! that removed a few lines of code from an already overgrown PR